zerver/views/users.py

```diff
@@ -468,24 +468,27 @@ def regenerate_bot_api_key(
 
 
 @require_member_or_admin
-@has_request_variables
+@typed_endpoint
 def add_bot_backend(
     request: HttpRequest,
     user_profile: UserProfile,
-    full_name_raw: str = REQ("full_name"),
-    short_name_raw: str = REQ("short_name"),
-    bot_type: int = REQ(json_validator=check_int, default=UserProfile.DEFAULT_BOT),
-    payload_url: str = REQ(json_validator=check_url, default=""),
-    service_name: Optional[str] = REQ(default=None),
-    config_data: Mapping[str, str] = REQ(
-        default={}, json_validator=check_dict(value_validator=check_string)
-    ),
-    interface_type: int = REQ(json_validator=check_int, default=Service.GENERIC),
-    default_sending_stream_name: Optional[str] = REQ("default_sending_stream", default=None),
-    default_events_register_stream_name: Optional[str] = REQ(
-        "default_events_register_stream", default=None
-    ),
-    default_all_public_streams: Optional[bool] = REQ(json_validator=check_bool, default=None),
+    *,
+    full_name_raw: Annotated[str, ApiParamConfig("full_name")],
+    short_name_raw: Annotated[str, ApiParamConfig("short_name")],
+    bot_type: Json[int] = UserProfile.DEFAULT_BOT,
+    payload_url: Json[
+        Annotated[str, AfterValidator(lambda x: check_param_url("payload_url", x))]
+    ] = "",
+    service_name: Optional[str] = None,
+    config_data: Json[Mapping[str, str]] = MappingProxyType({}),
```

Comment:
Why is `MappingProxyType` the right declaration here? It seems like a fairly obscure detail.

Reply:
Ruff doesn't allow function default arguments to be mutable. `MappingProxyType` creates a immutable proxy of the map passed to it.
An alternative would be to set the default to `None` and reassign to an empty `Dict` in the function itself.

Reply:
Yeah let's just default to `None` and reassign if needed.

Reply:
Ok, I've made the changes.